src/scripts/commands/database.ts

```diff
@@ -0,0 +1,205 @@
+DELETE FROM categoryoption_organisationunits     WHERE organisationunitid      IN (SELECT * FROM orgs);
+DELETE FROM chart_organisationunits              WHERE organisationunitid      IN (SELECT * FROM orgs);
+DELETE FROM completedatasetregistration          WHERE sourceid                IN (SELECT * FROM orgs);
+DELETE FROM configuration                        WHERE selfregistrationorgunit IN (SELECT * FROM orgs);
+DELETE FROM dataapproval                         WHERE organisationunitid      IN (SELECT * FROM orgs);
+DELETE FROM dataapprovalaudit                    WHERE organisationunitid      IN (SELECT * FROM orgs);
+DELETE FROM datasetsource                        WHERE sourceid                IN (SELECT * FROM orgs);
+DELETE FROM datavalue                            WHERE sourceid                IN (SELECT * FROM orgs);
+DELETE FROM datavalueaudit                       WHERE organisationunitid      IN (SELECT * FROM orgs);
+DELETE FROM eventchart_organisationunits         WHERE organisationunitid      IN (SELECT * FROM orgs);
+DELETE FROM eventreport_organisationunits        WHERE organisationunitid      IN (SELECT * FROM orgs);
+DELETE FROM eventvisualization_organisationunits WHERE organisationunitid      IN (SELECT * FROM orgs);
+DELETE FROM lockexception                        WHERE organisationunitid      IN (SELECT * FROM orgs);
+DELETE FROM mapview_organisationunits            WHERE organisationunitid      IN (SELECT * FROM orgs);
+DELETE FROM minmaxdataelement                    WHERE sourceid                IN (SELECT * FROM orgs);
+DELETE FROM organisationunitattributevalues      WHERE organisationunitid      IN (SELECT * FROM orgs);
+DELETE FROM orgunitgroupmembers                  WHERE organisationunitid      IN (SELECT * FROM orgs);
+DELETE FROM program_organisationunits            WHERE organisationunitid      IN (SELECT * FROM orgs);
+DELETE FROM programownershiphistory              WHERE organisationunitid      IN (SELECT * FROM orgs);
+DELETE FROM reporttable_organisationunits        WHERE organisationunitid      IN (SELECT * FROM orgs);
+DELETE FROM userdatavieworgunits                 WHERE organisationunitid      IN (SELECT * FROM orgs);
+DELETE FROM usermembership                       WHERE organisationunitid      IN (SELECT * FROM orgs);
+DELETE FROM userteisearchorgunits                WHERE organisationunitid      IN (SELECT * FROM orgs);
+DELETE FROM validationresult                     WHERE organisationunitid      IN (SELECT * FROM orgs);
+DELETE FROM visualization_organisationunits      WHERE organisationunitid      IN (SELECT * FROM orgs);
+
+
+\\echo -- Deleting orgunits themselves...
+
+DELETE FROM organisationunit WHERE organisationunitid IN (SELECT * FROM orgs);
+
+
+\\echo -- Cleaning up...
+
+DROP MATERIALIZED VIEW orgs CASCADE;`;
+}
+// Maybe this is worth checking too:
+// https://github.com/dhis2/dhis2-utils/blob/master/resources/sql/delete_orgunittree_with_data.sql
```

Comment:
What's the target DHIS2 version? Tried the script, it didn't find some of the tables (2.36.8):

```
$ node dist/index.js database remove-orgunit-level 3
```

Output: http://download.zaudera.com/public/remove_orgunits.sql.out

Reply:
Actually, it worked :) The output shows it selected 0 org units, and deleted 0 org units. The only errors are related to some tables not existing, which is okay, and you can see a previous output text warning that it may happen and it is still okay (that is related to other tables existing in other dhis2 versions).

I have tested it with several DBs with 2.36 and 2.38.

Reply:
I see, although the org unit is deleted from the DB, the API (this Maintenance) still show it. I guess that's normal, because of the cache and we need to restart the instance.